### DISCUSSION TITLE
Fixed IP to MAC detection (willing to pay)

### DISCUSSION THREAD

**Discussion by jck5000** [mletenay/home-assistant-goodwe-inverter#297] [Ideas]
Apologies, I have zero clue about github etiquette. The goodwe integration works perfectly fine, until either the inverter or my horrible, unchangeable hybrid modem (from only provider available) messes that fixed IP up.

_It would be really, really practical to auto-detect the IP based on MAC adress set up with first IP._

I was asked by my stepdad to find out if someone would do that if paid, and I don't know where else to ask. You _made_ that great stuff. He hadn't even heard about Python until I intruduced him to HA. And he was a programmer until his retirement; he's now 80yo. So it's up to me to ask.... so I do. Again, apologies if that's totally inappropriate, but I'm a bit despararate here.

**Comment by BenBarat** [comment]
Why not add your inverter with a static IP address on your router based on the MAC?
Then you can use that consistent IP address in your software configuration.
